### PR TITLE
move together int128 headers and fix neg int128 cast to floating numbers

### PR DESCRIPTION
**Pull request by Ashleyhx**

### CODE REVIEW

File: src/include/common/types/int128_t.h [hunk 1]
```diff
@@ -6,13 +6,51 @@
...
+#include "common/string_utils.h"
 
-namespace kuzu {
-namespace common {
+namespace kuzu::common {
 
 struct int128_t;
 
+// System representation for int128_t.
+struct KUZU_API int128_t {
```
Comment (acquamarin) [inline src/include/common/types/int128_t.h:17]: move KUZU_API before struct

File: src/include/common/types/int128_t.h [hunk 2]
```diff
@@ -6,13 +6,51 @@
 
 #include <stdint.h>
 
-#include "common/int128_t.h"
+#include "common/api.h"
+#include "common/string_utils.h"
 
-namespace kuzu {
-namespace common {
+namespace kuzu::common {
```
Comment (acquamarin) [inline src/include/common/types/int128_t.h:12]: Don't use nested namespace

File: src/common/types/int128_t.cpp [hunk 3]
```diff
@@ -425,7 +423,7 @@ template<class REAL_T>
 bool CastInt128ToFloating(int128_t input, REAL_T& result) {
     switch (input.high) {
     case -1:
-        result = -REAL_T(std::numeric_limits<REAL_T>::max() - input.low) - 1;
+        result = -REAL_T(std::numeric_limits<uint64_t>::max() - input.low) - 1;
```
Comment (acquamarin) [inline src/common/types/int128_t.cpp:426]: Use function::NumericLimits instead of std::numeric_limits.

File: src/common/types/int128_t.cpp [hunk 4]
```diff
@@ -425,7 +423,7 @@ template<class REAL_T>
...
         break;
     default:
         result = REAL_T(input.high) * std::numeric_limits<uint64_t>::max() + REAL_T(input.low);
```
Comment (acquamarin) [inline src/common/types/int128_t.cpp:429]: Ditto, use function::NumericLimits

File: src/common/types/int128_t.cpp [hunk 5]
```diff
@@ -619,5 +617,4 @@ int128_t::operator int64_t() const {
     return NarrowCast<int64_t>(*this);
 }
 
-} // namespace common
-} // namespace kuzu
+} // namespace kuzu::common
```
Comment (acquamarin) [inline src/common/types/int128_t.cpp:620]: Remove nested namespace

File: test/test_files/tinysnb/function/cast.test [hunk 6]
```diff
@@ -575,3 +574,12 @@ False
 ---- 1
 170141183460469231731687303715884105728.000000|-4324324
 
+-LOG CastNegInt128ToFloatDouble
```
Comment (acquamarin) [inline test/test_files/tinysnb/function/cast.test:577]: Can you confirm whether we have test cases:
uint->double/float, double/flat->uint

**Review by acquamarin — APPROVED** [review]

File: test/test_files/tinysnb/function/cast.test (hunk 6 above)
Comment (Ashleyhx) [reply test/test_files/tinysnb/function/cast.test:582]: we have `LOG CastUnsignedIntToFLoatDouble` and `-LOG CastDoubleToUInt`

**Pull request merged**